src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -244,6 +246,231 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
+                        }
+                    }
+                }
+            }
+        }
+
+        public ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> ProcessAnalyzerConfigFiles(
+            ImmutableArray<string> analyzerConfigPaths,
+            ImmutableArray<CommandLineSourceFile> sourceFiles,
+            ref bool hadErrors,
+            DiagnosticBag diagnostics)
+        {
+            var analyzerConfigs = ArrayBuilder<EditorConfig>.GetInstance();
+
+            readEditorConfigFiles(analyzerConfigPaths, analyzerConfigs);
+
+            if (diagnostics.HasAnyErrors())
+            {
+                analyzerConfigs.Free();
+                hadErrors = true;
+                return default;
+            }
+
+            var sourcePaths = sourceFiles.SelectAsArray(file => file.Path);
+            ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> diagnosticOptions =
+                GetAnalyzerConfigOptions(sourcePaths, analyzerConfigs, MessageProvider, diagnostics);
+
+            analyzerConfigs.Free();
+            return diagnosticOptions;
+
+            void readEditorConfigFiles(
+                ImmutableArray<string> configPaths,
+                ArrayBuilder<EditorConfig> configs)
+            {
+                var processedDirs = PooledHashSet<string>.GetInstance();
+
+                foreach (var configPath in configPaths)
+                {
+                    // The editorconfig spec requires all paths use '/' as the directory separator.
+                    // Since no known system allows directory separators as part of the file name,
+                    // we can replace every instance of the directory separator with a '/'
+                    string fileContent = TryReadFileContent(configPath, diagnostics, out string normalizedPath);
+                    if (fileContent is null)
+                    {
+                        // Error reading a file. Bail out and report error.
+                        break;
+                    }
+
+                    var directory = Path.GetDirectoryName(normalizedPath) ?? normalizedPath;
+                    // Path mapping expects a trailing '/' to be included
+                    if (directory[directory.Length - 1] != '/')
+                    {
+                        directory = directory + "/";
+                    }
+
+                    if (processedDirs.Contains(directory))
+                    {
+                        diagnostics.Add(Diagnostic.Create(
+                            MessageProvider,
+                            MessageProvider.ERR_MultipleAnalyzerConfigsInSameDir,
+                            directory));
+                        break;
+                    }
+
+                    processedDirs.Add(directory);
+                    var forwardPath = PathWithForwardSlashSeparators(directory);
+                    var editorConfig = EditorConfig.Parse(fileContent, forwardPath);
```

Comment:
How do errors get reported on this path?

Reply:
There are never any parse errors for EditorConfig files. Any invalid sections are skipped and no errors are reported.